File: frontend/src/service/authService.js
import api from './api';

export const login          = (data)        => api.post('/auth/login', data);
export const signup         = (data)        => api.post('/auth/signup', data);
export const forgotPassword = (email)       => api.post('/auth/forgot-password', { email });
export const resetPassword  = (data)        => api.post('/auth/reset-password', data);
export const getUserProfile = (id)          => api.get(`/users/${id}`);
export const updateUserProfile = (id, data) => api.put(`/users/${id}`, data);

File: Backend/src/main/java/StackVerse/Backend/service/AuthService.java
package StackVerse.Backend.service;

import StackVerse.Backend.dto.AuthRequestDTO;
import StackVerse.Backend.dto.AuthResponseDTO;
import StackVerse.Backend.dto.UserDTO;

public interface AuthService {
    AuthResponseDTO login(AuthRequestDTO request);
    UserDTO signup(UserDTO userDTO, String password);
    void forgotPassword(String email);
    void resetPassword(String email, String otp, String newPassword);
}
